### PR TITLE
Bug 1335060 - Extend/reorganize Screengraph nodes 

### PR DESCRIPTION
**Pull request by npark-mozilla**

### CODE REVIEW

**Comment by npark-mozilla** [comment]
aha, you can add paths that way too, thought it's allowed only within screengraph.  Then i don't have to set URL separately now, will do, thanks!

File: XCUITests/FxScreenGraph.swift [hunk 1]
```diff
@@ -186,3 +186,14 @@ func createScreenGraph(app: XCUIApplication) -> ScreenGraph {
 
     return map
 }
+
+extension Navigator {
+    func openNewURL(urlString: String) {
+        let newTabScreen = map.namedScenes[NewTabScreen]!
+        let app = map.app
+        newTabScreen.addEdge(BrowserTab) { _ in
+            app.textFields["url"].tap()
```
Comment (npark-mozilla) [inline XCUITests/FxScreenGraph.swift]: @jhugman debugger is telling me that the inside of {} is skipped when I call it from the script- is there a scope issue?  can't seem to figure out why (I'm new to swift as well)

Comment (jhugman) [inline XCUITests/FxScreenGraph.swift]: This second approach seems the simplest/most elegant – you don't need to make any changes to `ScreenGraph` at all then. I'd keep a route open the `BrowserTab` by having a default URL, just so other tests can get to it without thinking about it.

File: XCUITests/ScreenGraph.swift [hunk 2]
```diff
@@ -390,3 +390,4 @@ class Navigator {
         }
     }
 }
+
```
Comment (jhugman) [inline XCUITests/ScreenGraph.swift]: Nit: whitespace

File: XCUITests/FxScreenGraph.swift [hunk 3]
```diff
@@ -141,6 +142,9 @@ func createScreenGraph(app: XCUIApplication, url: String = "https://www.mozilla.
         scene.gesture(to: TabTrayMenu) {
             app.buttons["TabTrayController.menuButton"].tap()
         }
+        scene.gesture(to: NewTabScreen) {
+			app.buttons["TabTrayController.addTabButton"].tap()
+		}
```
Comment (jhugman) [inline XCUITests/FxScreenGraph.swift]: Nit: whitespace

File: XCUITests/BookmarkingTests.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		navigator.nowAt(BrowserTab)
+		checkBookmarked()
+		
+		// Open it, then unbookmark it, and check it's no longer on bookmarks home panel
+		unbookmark()
+		checkUnbookmarked()
+		
+    }
+    
+}
```
Comment (jhugman) [inline XCUITests/BookmarkingTests.swift]: Nits: whitespace.

Indent is four spaces. I think you have some tabs mixed in there too.

File: XCUITests/FxScreenGraph.swift [hunk 5]
```diff
@@ -172,10 +176,28 @@ func createScreenGraph(app: XCUIApplication, url: String = "https://www.mozilla.
...
+// Invoking this method in BrowserTab will create another tab, 
+// as that is the shortest path to itself
+extension Navigator {
+    func openNewURL(urlString: String) {
+        self.goto(NewTabScreen)
+        let app = XCUIApplication()
+        app.textFields["url"].tap()
+        app.textFields["address"].typeText(urlString + "\r")
+        self.nowAt(BrowserTab)
+    }
```
Comment (jhugman) [inline XCUITests/FxScreenGraph.swift]: YAAS! Good use of `nowAt`!

**Comment by npark-mozilla** [comment]
Thanks! now waiting for the swift 3 migration to complete before rebase/checking this in

**Pull request merged**